packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3187,6 +3188,13 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   final Queue<_NavigatorObservation> _observedRouteAdditions = Queue<_NavigatorObservation>();
   final Queue<_NavigatorObservation> _observedRouteDeletions = Queue<_NavigatorObservation>();
 
+  /// The [PrimaryScrollController] from current State. It is useful to access
+  /// root PrimaryScrollController in deeply nested Navigator.
+  /// ```dart
+  /// Navigator.of(context, rootNavigator: true).PrimaryScrollController
+  /// ```
```

Comment:
I'm not native english speaker, so if this description isn't clear, please update this comments.